From edab9fcc4515d4afe450e354c79a0a61786b7f46 Mon Sep 17 00:00:00 2001
From: Shammah Chancellor <shammah.chancellor@notvalid.com>
Date: Mon, 19 Oct 2026 06:46:15 -0700
Subject: [PATCH] test: declare the p2p tests' parameters as class attributes

Applies contrib/devtools/declarative-test-params.py to the p2p_* tests:
44 of 45 converted. p2p_headers_sync_with_minchainwork.py computes its
parameters and keeps set_test_params().

Every converted module imports, and the full functional suite gives the
same per-test results as before.
---
 test/functional/p2p_1p1c_network.py           | 11 +++----
 test/functional/p2p_add_connections.py        | 17 +++++-----
 test/functional/p2p_addr_relay.py             |  5 ++-
 test/functional/p2p_addrfetch.py              |  5 ++-
 test/functional/p2p_addrv2_relay.py           |  7 ++---
 test/functional/p2p_block_sync.py             |  5 ++-
 test/functional/p2p_blockfilters.py           | 15 +++++----
 test/functional/p2p_blocksonly.py             |  5 ++-
 test/functional/p2p_compactblocks.py          |  9 +++---
 .../p2p_compactblocks_blocksonly.py           |  5 ++-
 test/functional/p2p_compactblocks_hb.py       |  5 ++-
 test/functional/p2p_disconnect_ban.py         |  5 ++-
 test/functional/p2p_dos_header_tree.py        |  9 +++---
 test/functional/p2p_eviction.py               | 31 +++++++++----------
 test/functional/p2p_feefilter.py              | 27 ++++++++--------
 test/functional/p2p_filter.py                 |  7 ++---
 test/functional/p2p_fingerprint.py            |  5 ++-
 test/functional/p2p_getaddr_caching.py        |  9 +++---
 test/functional/p2p_getdata.py                |  3 +-
 test/functional/p2p_i2p_ports.py              |  7 ++---
 test/functional/p2p_i2p_sessions.py           | 13 ++++----
 test/functional/p2p_ibd_stalling.py           |  5 ++-
 test/functional/p2p_ibd_txrelay.py            | 13 ++++----
 test/functional/p2p_initial_headers_sync.py   |  5 ++-
 test/functional/p2p_inv_download.py           | 17 +++++-----
 test/functional/p2p_invalid_block.py          |  7 ++---
 test/functional/p2p_invalid_locator.py        |  3 +-
 test/functional/p2p_invalid_messages.py       |  7 ++---
 test/functional/p2p_invalid_tx.py             | 13 ++++----
 test/functional/p2p_leak.py                   |  3 +-
 test/functional/p2p_leak_tx.py                | 29 +++++++++--------
 test/functional/p2p_message_capture.py        |  7 ++---
 test/functional/p2p_mutated_blocks.py         |  5 ++-
 test/functional/p2p_net_deadlock.py           |  3 +-
 test/functional/p2p_nobloomfilter_messages.py |  7 ++---
 test/functional/p2p_node_network_limited.py   |  7 ++---
 test/functional/p2p_opportunistic_1p1c.py     | 17 +++++-----
 test/functional/p2p_orphan_handling.py        |  5 ++-
 test/functional/p2p_permissions.py            |  5 ++-
 test/functional/p2p_ping.py                   | 11 +++----
 test/functional/p2p_sendheaders.py            |  7 ++---
 test/functional/p2p_timeouts.py               |  9 +++---
 test/functional/p2p_tx_privacy.py             |  3 +-
 test/functional/p2p_unrequested_blocks.py     |  7 ++---
 44 files changed, 178 insertions(+), 222 deletions(-)

diff --git a/test/functional/p2p_1p1c_network.py b/test/functional/p2p_1p1c_network.py
index 1a8fb0474528292efca70c3ea1f8fa8246a28961..99072e24d3424d764f58267fe834f85a6e588b76 100644
--- a/test/functional/p2p_1p1c_network.py
+++ b/test/functional/p2p_1p1c_network.py
@@ -22,12 +22,11 @@ FEERATE_1SAT_VB = Decimal("10")
 
 
 class PackageRelayTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 4
-        self.noban_tx_relay = True
-        self.extra_args = [["-acceptnonstdtxn=1", "-maxmempool=5"]] * self.num_nodes
-        self.supports_cli = False
+    setup_clean_chain = True
+    num_nodes = 4
+    noban_tx_relay = True
+    extra_args = [["-acceptnonstdtxn=1", "-maxmempool=5"]] * num_nodes
+    supports_cli = False
 
     def raise_network_minfee(self):
         filler_wallet = MiniWallet(self.nodes[0])
diff --git a/test/functional/p2p_add_connections.py b/test/functional/p2p_add_connections.py
index 2fac166c55e9d4570dda2da7e7702bdaa3fb07e0..7db92b0219303acded0e5cc4b0000d26953e13a3 100644
--- a/test/functional/p2p_add_connections.py
+++ b/test/functional/p2p_add_connections.py
@@ -29,15 +29,14 @@ class P2PFeelerReceiver(P2PInterface):
 
 
 class P2PAddConnections(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 2
-        self.extra_args = [
-            [
-                "-avaproofstakeutxoconfirmations=1",
-                f"-maxavalancheoutbound={MAX_AVALANCHE_OUTBOUND_CONNECTIONS}",
-            ],
-            [],
-        ]
+    num_nodes = 2
+    extra_args = [
+        [
+            "-avaproofstakeutxoconfirmations=1",
+            f"-maxavalancheoutbound={MAX_AVALANCHE_OUTBOUND_CONNECTIONS}",
+        ],
+        [],
+    ]
 
     def setup_network(self):
         self.setup_nodes()
diff --git a/test/functional/p2p_addr_relay.py b/test/functional/p2p_addr_relay.py
index fd6c018b393fdf725280e4c6bf0e43bbb888a493..3b4a8766fa04016d64ef735ad2ec35e7f05a468a 100644
--- a/test/functional/p2p_addr_relay.py
+++ b/test/functional/p2p_addr_relay.py
@@ -88,9 +88,8 @@ class AddrTest(BitcoinTestFramework):
     counter = 0
     mocktime = int(time.time())
 
-    def set_test_params(self):
-        self.num_nodes = 1
-        self.extra_args = [["-whitelist=addr@127.0.0.1"]]
+    num_nodes = 1
+    extra_args = [["-whitelist=addr@127.0.0.1"]]
 
     def run_test(self):
         self.oversized_addr_test()
diff --git a/test/functional/p2p_addrfetch.py b/test/functional/p2p_addrfetch.py
index 2d594877c497ea1af88f86b485a9f5f2f1d3b1f0..e110c51e82ca4e3978428a381ccff1664d4b153d 100644
--- a/test/functional/p2p_addrfetch.py
+++ b/test/functional/p2p_addrfetch.py
@@ -20,9 +20,8 @@ ADDR.port = 18444
 
 
 class P2PAddrFetch(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 1
+    setup_clean_chain = True
+    num_nodes = 1
 
     def assert_getpeerinfo(self, *, peer_ids):
         num_peers = len(peer_ids)
diff --git a/test/functional/p2p_addrv2_relay.py b/test/functional/p2p_addrv2_relay.py
index 6ba1a3349048ce59970f8ff6f42a9613c983b784..0b6b18b002edf0dc49828a055eae92c231513924 100644
--- a/test/functional/p2p_addrv2_relay.py
+++ b/test/functional/p2p_addrv2_relay.py
@@ -38,10 +38,9 @@ class AddrReceiver(P2PInterface):
 
 
 class AddrTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 1
-        self.extra_args = [["-whitelist=addr@127.0.0.1"]]
+    setup_clean_chain = True
+    num_nodes = 1
+    extra_args = [["-whitelist=addr@127.0.0.1"]]
 
     def run_test(self):
         self.log.info("Create connection that sends addrv2 messages")
diff --git a/test/functional/p2p_block_sync.py b/test/functional/p2p_block_sync.py
index 853f6be88b0bc4a0ff300a311e8013f1661d64b9..40432d9dd2a010ea48ac2c4b83485346aa282a51 100644
--- a/test/functional/p2p_block_sync.py
+++ b/test/functional/p2p_block_sync.py
@@ -11,9 +11,8 @@ from test_framework.test_framework import BitcoinTestFramework
 
 
 class BlockSyncTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 3
+    setup_clean_chain = True
+    num_nodes = 3
 
     def setup_network(self):
         self.setup_nodes()
diff --git a/test/functional/p2p_blockfilters.py b/test/functional/p2p_blockfilters.py
index 035f283289d255309c934653b0fd9d99615534af..ee2a2eea39b05332f37be4cf20f940048ebcd997 100644
--- a/test/functional/p2p_blockfilters.py
+++ b/test/functional/p2p_blockfilters.py
@@ -39,14 +39,13 @@ class FiltersClient(P2PInterface):
 
 
 class CompactFiltersTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.rpc_timeout = 480
-        self.num_nodes = 2
-        self.extra_args = [
-            ["-blockfilterindex", "-peerblockfilters"],
-            ["-blockfilterindex"],
-        ]
+    setup_clean_chain = True
+    rpc_timeout = 480
+    num_nodes = 2
+    extra_args = [
+        ["-blockfilterindex", "-peerblockfilters"],
+        ["-blockfilterindex"],
+    ]
 
     def run_test(self):
         # Node 0 supports COMPACT_FILTERS, node 1 does not.
diff --git a/test/functional/p2p_blocksonly.py b/test/functional/p2p_blocksonly.py
index 65319840bebe11f3ba3a08646160f85d6179574d..b2dfe967e6afccf5f09f27a6f438d72b659fd6d7 100644
--- a/test/functional/p2p_blocksonly.py
+++ b/test/functional/p2p_blocksonly.py
@@ -13,9 +13,8 @@ from test_framework.wallet import MiniWallet
 
 
 class P2PBlocksOnly(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
-        self.extra_args = [["-blocksonly"]]
+    num_nodes = 1
+    extra_args = [["-blocksonly"]]
 
     def run_test(self):
         self.miniwallet = MiniWallet(self.nodes[0])
diff --git a/test/functional/p2p_compactblocks.py b/test/functional/p2p_compactblocks.py
index 2ad0dea39d035a535dc5e65b685d88c5b5346b06..1eb2d351d9188f56995b280d9b49ef964258eb09 100644
--- a/test/functional/p2p_compactblocks.py
+++ b/test/functional/p2p_compactblocks.py
@@ -133,11 +133,10 @@ class TestP2PConn(P2PInterface):
 
 
 class CompactBlocksTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 1
-        self.extra_args = [["-acceptnonstdtxn=1"]]
-        self.utxos = []
+    setup_clean_chain = True
+    num_nodes = 1
+    extra_args = [["-acceptnonstdtxn=1"]]
+    utxos = []
 
     def skip_test_if_missing_module(self):
         self.skip_if_no_wallet()
diff --git a/test/functional/p2p_compactblocks_blocksonly.py b/test/functional/p2p_compactblocks_blocksonly.py
index 0cd20b002e1e560143c3bc79b765bf8cbf130e42..6534e8d6b918ebfce47efd32adb2dba3203afd99 100644
--- a/test/functional/p2p_compactblocks_blocksonly.py
+++ b/test/functional/p2p_compactblocks_blocksonly.py
@@ -21,9 +21,8 @@ from test_framework.util import assert_equal
 
 
 class P2PCompactBlocksBlocksOnly(BitcoinTestFramework):
-    def set_test_params(self):
-        self.extra_args = [["-blocksonly"], [], [], []]
-        self.num_nodes = 4
+    extra_args = [["-blocksonly"], [], [], []]
+    num_nodes = 4
 
     def setup_network(self):
         self.setup_nodes()
diff --git a/test/functional/p2p_compactblocks_hb.py b/test/functional/p2p_compactblocks_hb.py
index 5ef8d7e51ca19a4cde651f2b07b24fc95ad130bf..fdb69e330a264328746328946eb51a0640be1464 100644
--- a/test/functional/p2p_compactblocks_hb.py
+++ b/test/functional/p2p_compactblocks_hb.py
@@ -10,9 +10,8 @@ from test_framework.util import assert_equal
 class CompactBlocksConnectionTest(BitcoinTestFramework):
     """Test class for verifying selection of HB peer connections."""
 
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 6
+    setup_clean_chain = True
+    num_nodes = 6
 
     def peer_info(self, from_node, to_node):
         """Query from_node for its getpeerinfo about to_node."""
diff --git a/test/functional/p2p_disconnect_ban.py b/test/functional/p2p_disconnect_ban.py
index dced4863f7422b34e9525eca4740968e19dec256..9dea34f8dfcde18f2ed2f022db490c75a9d7cd06 100644
--- a/test/functional/p2p_disconnect_ban.py
+++ b/test/functional/p2p_disconnect_ban.py
@@ -10,9 +10,8 @@ from test_framework.util import assert_equal, assert_raises_rpc_error
 
 
 class DisconnectBanTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 2
-        self.supports_cli = False
+    num_nodes = 2
+    supports_cli = False
 
     def run_test(self):
         self.log.info("Connect nodes both way")
diff --git a/test/functional/p2p_dos_header_tree.py b/test/functional/p2p_dos_header_tree.py
index 5fbadac33ff3d5452f4aa638483beadbbb995ab1..8f01adeebf3118d6de76b29721ba63aff2bfc6ea 100644
--- a/test/functional/p2p_dos_header_tree.py
+++ b/test/functional/p2p_dos_header_tree.py
@@ -11,11 +11,10 @@ from test_framework.test_framework import BitcoinTestFramework
 
 
 class RejectLowDifficultyHeadersTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.chain = "testnet3"  # Use testnet chain because it has an early checkpoint
-        self.num_nodes = 2
-        self.extra_args = [["-minimumchainwork=0x0"], ["-minimumchainwork=0x0"]]
+    setup_clean_chain = True
+    chain = "testnet3"  # Use testnet chain because it has an early checkpoint
+    num_nodes = 2
+    extra_args = [["-minimumchainwork=0x0"], ["-minimumchainwork=0x0"]]
 
     def add_options(self, parser):
         parser.add_argument(
diff --git a/test/functional/p2p_eviction.py b/test/functional/p2p_eviction.py
index b1a15098cf8ce8d1a616fd82423ec35613e69344..7af87add51efacb9e1d25ef6966d9293813cd6bb 100644
--- a/test/functional/p2p_eviction.py
+++ b/test/functional/p2p_eviction.py
@@ -55,23 +55,22 @@ class SlowAvaP2PInterface(AvaP2PInterface):
 
 
 class P2PEvict(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
-        # The choice of maxconnections=188 results in a maximum of 153 inbound
-        # connections (188 - 34 outbound - 1 feeler). The 34 outbounds count is
-        # from 16 full-relay + 16 avalanche + 2 block-only-relay.
-        # 152 inbound peers are protected from eviction:
-        # 4 by netgroup, 4 that sent us blocks, 4 that sent us proofs, 4 that
-        # sent us transactions, 8 via lowest ping time, 128 with the best
-        # avalanche availability score
-        self.extra_args = [
-            [
-                "-maxconnections=188",
-                "-avaproofstakeutxodustthreshold=1000000",
-                "-avaproofstakeutxoconfirmations=1",
-                "-maxavalancheoutbound=16",
-            ]
+    num_nodes = 1
+    # The choice of maxconnections=188 results in a maximum of 153 inbound
+    # connections (188 - 34 outbound - 1 feeler). The 34 outbounds count is
+    # from 16 full-relay + 16 avalanche + 2 block-only-relay.
+    # 152 inbound peers are protected from eviction:
+    # 4 by netgroup, 4 that sent us blocks, 4 that sent us proofs, 4 that
+    # sent us transactions, 8 via lowest ping time, 128 with the best
+    # avalanche availability score
+    extra_args = [
+        [
+            "-maxconnections=188",
+            "-avaproofstakeutxodustthreshold=1000000",
+            "-avaproofstakeutxoconfirmations=1",
+            "-maxavalancheoutbound=16",
         ]
+    ]
 
     def run_test(self):
         # peers that we expect to be protected from eviction
diff --git a/test/functional/p2p_feefilter.py b/test/functional/p2p_feefilter.py
index 8c1f2724378539ac8d9e3a17819b7055971f55ae..ce9768b8103292e2c8fffd45591c3796b68a66bb 100644
--- a/test/functional/p2p_feefilter.py
+++ b/test/functional/p2p_feefilter.py
@@ -43,20 +43,19 @@ class TestP2PConn(P2PInterface):
 
 
 class FeeFilterTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 2
-        self.noban_tx_relay = True
-        # We lower the various required feerates for this test
-        # to catch a corner-case where feefilter used to slightly undercut
-        # mempool and wallet feerate calculation based on GetFee
-        # rounding down 3 places, leading to stranded transactions.
-        # See issue #16499
-        self.extra_args = [
-            [
-                "-minrelaytxfee=1",
-                "-mintxfee=1",
-            ]
-        ] * self.num_nodes
+    num_nodes = 2
+    noban_tx_relay = True
+    # We lower the various required feerates for this test
+    # to catch a corner-case where feefilter used to slightly undercut
+    # mempool and wallet feerate calculation based on GetFee
+    # rounding down 3 places, leading to stranded transactions.
+    # See issue #16499
+    extra_args = [
+        [
+            "-minrelaytxfee=1",
+            "-mintxfee=1",
+        ]
+    ] * num_nodes
 
     def run_test(self):
         self.test_feefilter_forcerelay()
diff --git a/test/functional/p2p_filter.py b/test/functional/p2p_filter.py
index 09b915ea02583664a368067057db13b116f22c30..a50fd61fd3374de72290995e5c99961bf0fd43de 100644
--- a/test/functional/p2p_filter.py
+++ b/test/functional/p2p_filter.py
@@ -107,10 +107,9 @@ class P2PBloomFilter(P2PInterface):
 
 
 class FilterTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
-        self.noban_tx_relay = True
-        self.extra_args = [["-peerbloomfilters"]]
+    num_nodes = 1
+    noban_tx_relay = True
+    extra_args = [["-peerbloomfilters"]]
 
     def generatetoscriptpubkey(self, scriptpubkey):
         """Helper to generate a single block to the given scriptPubKey."""
diff --git a/test/functional/p2p_fingerprint.py b/test/functional/p2p_fingerprint.py
index 37ab9170f368b8f249a9cc10e0172c3829788815..6596565ee4c30a1d5d7685e9cd9ebc822500945e 100644
--- a/test/functional/p2p_fingerprint.py
+++ b/test/functional/p2p_fingerprint.py
@@ -24,9 +24,8 @@ from test_framework.util import assert_equal
 
 
 class P2PFingerprintTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 1
+    setup_clean_chain = True
+    num_nodes = 1
 
     # Build a chain of blocks on top of given one
     def build_chain(self, nblocks, prev_hash, prev_height, prev_median_time):
diff --git a/test/functional/p2p_getaddr_caching.py b/test/functional/p2p_getaddr_caching.py
index 5ba0e01a846be49396ee5af63b8bf07ef1d985cb..7f8d9bddfad1f17de854d222796584f0c7f00ccf 100644
--- a/test/functional/p2p_getaddr_caching.py
+++ b/test/functional/p2p_getaddr_caching.py
@@ -34,11 +34,10 @@ class AddrReceiver(P2PInterface):
 
 
 class AddrTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
-        # TODO When Backporting core#25096, please also includes changes from:
-        #  - core#25312
-        #  - core#25333
+    num_nodes = 1
+    # TODO When Backporting core#25096, please also includes changes from:
+    #  - core#25312
+    #  - core#25333
 
     def run_test(self):
         self.log.info("Fill peer AddrMan with a lot of records")
diff --git a/test/functional/p2p_getdata.py b/test/functional/p2p_getdata.py
index e07bfdd58ceb087965469233046356803bb57d9e..94410deefb1f382ea51ab3024ed627a4be7a3fa8 100644
--- a/test/functional/p2p_getdata.py
+++ b/test/functional/p2p_getdata.py
@@ -20,8 +20,7 @@ class P2PStoreBlock(P2PInterface):
 
 
 class GetdataTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
+    num_nodes = 1
 
     def run_test(self):
         p2p_block_store = self.nodes[0].add_p2p_connection(P2PStoreBlock())
diff --git a/test/functional/p2p_i2p_ports.py b/test/functional/p2p_i2p_ports.py
index 2d55c93628633f4c75987d814cfde012a8629b18..82ca8618581a36b1393e91a01f2da5f39add3ef0 100644
--- a/test/functional/p2p_i2p_ports.py
+++ b/test/functional/p2p_i2p_ports.py
@@ -11,10 +11,9 @@ from test_framework.test_framework import BitcoinTestFramework
 
 
 class I2PPorts(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
-        # The test assumes that an I2P SAM proxy is not listening here.
-        self.extra_args = [["-i2psam=127.0.0.1:60000"]]
+    num_nodes = 1
+    # The test assumes that an I2P SAM proxy is not listening here.
+    extra_args = [["-i2psam=127.0.0.1:60000"]]
 
     def run_test(self):
         node = self.nodes[0]
diff --git a/test/functional/p2p_i2p_sessions.py b/test/functional/p2p_i2p_sessions.py
index ab348dc40eb272f879592d3f25619e68c647db96..b89469801ca29df1c240c629e7b43b8a98841cba 100644
--- a/test/functional/p2p_i2p_sessions.py
+++ b/test/functional/p2p_i2p_sessions.py
@@ -9,13 +9,12 @@ from test_framework.test_framework import BitcoinTestFramework
 
 
 class I2PSessions(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 2
-        # The test assumes that an I2P SAM proxy is not listening here.
-        self.extra_args = [
-            ["-i2psam=127.0.0.1:60000", "-i2pacceptincoming=1"],
-            ["-i2psam=127.0.0.1:60000", "-i2pacceptincoming=0"],
-        ]
+    num_nodes = 2
+    # The test assumes that an I2P SAM proxy is not listening here.
+    extra_args = [
+        ["-i2psam=127.0.0.1:60000", "-i2pacceptincoming=1"],
+        ["-i2psam=127.0.0.1:60000", "-i2pacceptincoming=0"],
+    ]
 
     def run_test(self):
         addr = "zsxwyo6qcn3chqzwxnseusqgsnuw3maqnztkiypyfxtya4snkoka.b32.i2p"
diff --git a/test/functional/p2p_ibd_stalling.py b/test/functional/p2p_ibd_stalling.py
index bc111ba4e44e805afed9a40b5a344aa92d797af6..9ecee162f1d9881859c71c6f6b9d921b15ec1584 100644
--- a/test/functional/p2p_ibd_stalling.py
+++ b/test/functional/p2p_ibd_stalling.py
@@ -37,9 +37,8 @@ class P2PStaller(P2PDataStore):
 
 
 class P2PIBDStallingTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 1
+    setup_clean_chain = True
+    num_nodes = 1
 
     def run_test(self):
         NUM_BLOCKS = 1025
diff --git a/test/functional/p2p_ibd_txrelay.py b/test/functional/p2p_ibd_txrelay.py
index a671544688214260c756e69e1e2fbb8365504693..800320e6d4a45bbe6df92ffd323808eb4e9731e3 100644
--- a/test/functional/p2p_ibd_txrelay.py
+++ b/test/functional/p2p_ibd_txrelay.py
@@ -32,13 +32,12 @@ NORMAL_FEE_FILTER = Decimal(100) / XEC
 
 
 class P2PIBDTxRelayTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 2
-        self.extra_args = [
-            [f"-minrelaytxfee={NORMAL_FEE_FILTER}"],
-            [f"-minrelaytxfee={NORMAL_FEE_FILTER}"],
-        ]
+    setup_clean_chain = True
+    num_nodes = 2
+    extra_args = [
+        [f"-minrelaytxfee={NORMAL_FEE_FILTER}"],
+        [f"-minrelaytxfee={NORMAL_FEE_FILTER}"],
+    ]
 
     def run_test(self):
         self.log.info("Check that nodes set minfilter to MAX_MONEY while still in IBD")
diff --git a/test/functional/p2p_initial_headers_sync.py b/test/functional/p2p_initial_headers_sync.py
index 0ccc1dd0f4195f62d7c41da75919d53e6fa7c1c0..6ebf020c8a51c658d018260bcb9ba8113195feb9 100644
--- a/test/functional/p2p_initial_headers_sync.py
+++ b/test/functional/p2p_initial_headers_sync.py
@@ -17,9 +17,8 @@ from test_framework.util import assert_equal
 
 
 class HeadersSyncTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 1
+    setup_clean_chain = True
+    num_nodes = 1
 
     def announce_random_block(self, peers):
         new_block_announcement = msg_inv(
diff --git a/test/functional/p2p_inv_download.py b/test/functional/p2p_inv_download.py
index 8258eca8277db906c009d7f5a6a32ecf6a3ee604..cadada9f2152bf20f5b8f60d149d1df7d991c3bb 100644
--- a/test/functional/p2p_inv_download.py
+++ b/test/functional/p2p_inv_download.py
@@ -128,15 +128,14 @@ def skip(context):
 
 
 class InventoryDownloadTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 2
-        self.extra_args = [
-            [
-                "-avaproofstakeutxodustthreshold=1000000",
-                "-avaproofstakeutxoconfirmations=1",
-                "-avacooldown=0",
-            ]
-        ] * self.num_nodes
+    num_nodes = 2
+    extra_args = [
+        [
+            "-avaproofstakeutxodustthreshold=1000000",
+            "-avaproofstakeutxoconfirmations=1",
+            "-avacooldown=0",
+        ]
+    ] * num_nodes
 
     def test_data_requests(self, context):
         self.log.info("Test that we request data from all our peers, eventually")
diff --git a/test/functional/p2p_invalid_block.py b/test/functional/p2p_invalid_block.py
index ef83fc3c3e6e9a739d364777fabe80070519f26e..5566277d7698c9cb6d017458bb303b5c1319cfb0 100644
--- a/test/functional/p2p_invalid_block.py
+++ b/test/functional/p2p_invalid_block.py
@@ -31,10 +31,9 @@ from test_framework.util import assert_equal
 
 
 class InvalidBlockRequestTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
-        self.setup_clean_chain = True
-        self.noban_tx_relay = True
+    num_nodes = 1
+    setup_clean_chain = True
+    noban_tx_relay = True
 
     def run_test(self):
         # Add p2p connection to node0
diff --git a/test/functional/p2p_invalid_locator.py b/test/functional/p2p_invalid_locator.py
index cabeb87ff1fe8c2bcf555489c5e734d3f36ca885..3239255ba6df892c35341c62fc1e1830d1931ee8 100644
--- a/test/functional/p2p_invalid_locator.py
+++ b/test/functional/p2p_invalid_locator.py
@@ -9,8 +9,7 @@ from test_framework.test_framework import BitcoinTestFramework
 
 
 class InvalidLocatorTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
+    num_nodes = 1
 
     def run_test(self):
         # Convenience reference to the node
diff --git a/test/functional/p2p_invalid_messages.py b/test/functional/p2p_invalid_messages.py
index 8031e6bc2d9854bb7a32ef25bf4a2f260ccbe3be..5153dd3b25c86db1af2ab8cb28e3740062e4a8f1 100644
--- a/test/functional/p2p_invalid_messages.py
+++ b/test/functional/p2p_invalid_messages.py
@@ -55,10 +55,9 @@ class SenderOfAddrV2(P2PInterface):
 
 
 class InvalidMessagesTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
-        self.setup_clean_chain = True
-        self.extra_args = [["-whitelist=addr@127.0.0.1"]]
+    num_nodes = 1
+    setup_clean_chain = True
+    extra_args = [["-whitelist=addr@127.0.0.1"]]
 
     def run_test(self):
         self.test_buffer()
diff --git a/test/functional/p2p_invalid_tx.py b/test/functional/p2p_invalid_tx.py
index 0441a8e7a28fefda66e6d91361d104b46c9ddec7..025b1b6d13c2a9f14985ee80104c41bb4aeeeb06 100644
--- a/test/functional/p2p_invalid_tx.py
+++ b/test/functional/p2p_invalid_tx.py
@@ -22,14 +22,13 @@ from test_framework.util import assert_equal
 
 
 class InvalidTxRequestTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
-        self.extra_args = [
-            [
-                "-acceptnonstdtxn=1",
-            ]
+    num_nodes = 1
+    extra_args = [
+        [
+            "-acceptnonstdtxn=1",
         ]
-        self.setup_clean_chain = True
+    ]
+    setup_clean_chain = True
 
     def bootstrap_p2p(self, *, num_connections=1):
         """Add a P2P connection to the node.
diff --git a/test/functional/p2p_leak.py b/test/functional/p2p_leak.py
index 970bc49511ab89a4d459aa372edfed80aeb52389..8be7828443150114f42bf21133fc559d2aa72add 100644
--- a/test/functional/p2p_leak.py
+++ b/test/functional/p2p_leak.py
@@ -129,8 +129,7 @@ class P2PVersionStore(P2PInterface):
 
 
 class P2PLeakTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
+    num_nodes = 1
 
     def run_test(self):
         # Peer that never sends a version. We will send a bunch of messages
diff --git a/test/functional/p2p_leak_tx.py b/test/functional/p2p_leak_tx.py
index 2e0c26cbc4cb129527d72505f51b19bc5f5989aa..7a2e7a5554a96c6521ee057c26047a730ab5403b 100644
--- a/test/functional/p2p_leak_tx.py
+++ b/test/functional/p2p_leak_tx.py
@@ -21,21 +21,20 @@ class P2PNode(P2PDataStore):
 
 
 class P2PLeakTxTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
-        self.noban_tx_relay = True
-        self.extra_args = [
-            [
-                "-avacooldown=0",
-                # Keep the manual reorg parked until its explicit restoration.
-                "-automaticunparking=0",
-                "-avaproofstakeutxoconfirmations=1",
-                # Low enough for coinbase transactions to be staked in valid proofs
-                "-avaproofstakeutxodustthreshold=1000000",
-                "-avaminquorumstake=0",
-                "-avaminavaproofsnodecount=0",
-            ],
-        ]
+    num_nodes = 1
+    noban_tx_relay = True
+    extra_args = [
+        [
+            "-avacooldown=0",
+            # Keep the manual reorg parked until its explicit restoration.
+            "-automaticunparking=0",
+            "-avaproofstakeutxoconfirmations=1",
+            # Low enough for coinbase transactions to be staked in valid proofs
+            "-avaproofstakeutxodustthreshold=1000000",
+            "-avaminquorumstake=0",
+            "-avaminavaproofsnodecount=0",
+        ],
+    ]
 
     def run_test(self):
         # The block and tx generating node
diff --git a/test/functional/p2p_message_capture.py b/test/functional/p2p_message_capture.py
index fdceb0e5c03f0993a1b3139e83d8ff33c6c5f926..f9a9a7d74f81ec24c21ad5d905aca5b5b8d13eef 100644
--- a/test/functional/p2p_message_capture.py
+++ b/test/functional/p2p_message_capture.py
@@ -57,10 +57,9 @@ def mini_parser(dat_file):
 
 
 class MessageCaptureTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
-        self.extra_args = [["-capturemessages"]]
-        self.setup_clean_chain = True
+    num_nodes = 1
+    extra_args = [["-capturemessages"]]
+    setup_clean_chain = True
 
     def run_test(self):
         capturedir = os.path.join(self.nodes[0].datadir, "regtest/message_capture")
diff --git a/test/functional/p2p_mutated_blocks.py b/test/functional/p2p_mutated_blocks.py
index bd9fb19d2fdb30712a31f9d2054c875bcd5b71ef..617d21c2ba5fd70a9bfbfcccf3e9fc4eeee76419 100644
--- a/test/functional/p2p_mutated_blocks.py
+++ b/test/functional/p2p_mutated_blocks.py
@@ -24,9 +24,8 @@ from test_framework.wallet import MiniWallet
 
 
 class MutatedBlocksTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 1
+    setup_clean_chain = True
+    num_nodes = 1
 
     def run_test(self):
         self.wallet = MiniWallet(self.nodes[0])
diff --git a/test/functional/p2p_net_deadlock.py b/test/functional/p2p_net_deadlock.py
index f917f40b79f31bc666b4a15f580b664fd164be71..1e6059e6619c6f679ad29de4e0a7bbc8505dcab3 100644
--- a/test/functional/p2p_net_deadlock.py
+++ b/test/functional/p2p_net_deadlock.py
@@ -19,8 +19,7 @@ from test_framework.wallet import MiniWallet
 
 
 class NetDeadlockTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 2
+    num_nodes = 2
 
     def run_test(self):
         node0 = self.nodes[0]
diff --git a/test/functional/p2p_nobloomfilter_messages.py b/test/functional/p2p_nobloomfilter_messages.py
index 4966bf115f8d77596d42559620be6d9ff0a9e5a6..780d29d34018d14dd84fa7830ed2bff7d8c34561 100644
--- a/test/functional/p2p_nobloomfilter_messages.py
+++ b/test/functional/p2p_nobloomfilter_messages.py
@@ -22,10 +22,9 @@ from test_framework.util import assert_equal
 
 
 class P2PNoBloomFilterMessages(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 1
-        self.extra_args = [["-peerbloomfilters=0"]]
+    setup_clean_chain = True
+    num_nodes = 1
+    extra_args = [["-peerbloomfilters=0"]]
 
     def test_message_causes_disconnect(self, message):
         """Add a p2p connection that sends a message and check that it disconnects."""
diff --git a/test/functional/p2p_node_network_limited.py b/test/functional/p2p_node_network_limited.py
index a18b195b9c1e5da3c7baa99c9593074a16909b1f..2d6bfdc314f12ef9106d4525c6bf8115495d215d 100644
--- a/test/functional/p2p_node_network_limited.py
+++ b/test/functional/p2p_node_network_limited.py
@@ -45,10 +45,9 @@ class P2PIgnoreInv(P2PInterface):
 
 
 class NodeNetworkLimitedTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 3
-        self.extra_args = [["-prune=550", "-addrmantest"], [], []]
+    setup_clean_chain = True
+    num_nodes = 3
+    extra_args = [["-prune=550", "-addrmantest"], [], []]
 
     def disconnect_all(self):
         self.disconnect_nodes(0, 1)
diff --git a/test/functional/p2p_opportunistic_1p1c.py b/test/functional/p2p_opportunistic_1p1c.py
index cc6b5c5ec9e01d90706e4926817c8e388cdf4d48..bd542de98ecdd2160b2afde217ba16c739fe855e 100644
--- a/test/functional/p2p_opportunistic_1p1c.py
+++ b/test/functional/p2p_opportunistic_1p1c.py
@@ -45,16 +45,15 @@ def cleanup(func):
 
 
 class PackageRelayTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 1
-        self.extra_args = [
-            [
-                "-acceptnonstdtxn=1",
-                "-maxmempool=5",
-            ]
+    setup_clean_chain = True
+    num_nodes = 1
+    extra_args = [
+        [
+            "-acceptnonstdtxn=1",
+            "-maxmempool=5",
         ]
-        self.supports_cli = False
+    ]
+    supports_cli = False
 
     def create_tx_below_mempoolminfee(self, wallet):
         """Create a 1-input 1sat/vB transaction using a confirmed UTXO.
diff --git a/test/functional/p2p_orphan_handling.py b/test/functional/p2p_orphan_handling.py
index 3fe8b363fbba7a00186c285b7193576ed359c7bb..b03cadc55d77936f5e9813ef4e2bbf6d6c501888 100644
--- a/test/functional/p2p_orphan_handling.py
+++ b/test/functional/p2p_orphan_handling.py
@@ -109,9 +109,8 @@ class PeerTxRelayer(P2PTxInvStore):
 
 
 class OrphanHandlingTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
-        self.extra_args = [[]]
+    num_nodes = 1
+    extra_args = [[]]
 
     def create_parent_and_child(self):
         """Create package with 1 parent and 1 child, normal fees (no cpfp)."""
diff --git a/test/functional/p2p_permissions.py b/test/functional/p2p_permissions.py
index b9bb3e6b6d3b54abee4c2a87dafc0db5822e026d..503fb3ad69672812ae93e29fa91e3ac752a79194 100644
--- a/test/functional/p2p_permissions.py
+++ b/test/functional/p2p_permissions.py
@@ -17,9 +17,8 @@ from test_framework.util import append_config, assert_equal, p2p_port, tor_port
 
 
 class P2PPermissionsTests(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 2
-        self.setup_clean_chain = True
+    num_nodes = 2
+    setup_clean_chain = True
 
     def run_test(self):
         self.check_tx_relay()
diff --git a/test/functional/p2p_ping.py b/test/functional/p2p_ping.py
index bfdd446e869520442db16b052961164d1258df99..d4bcf55c789c2d64de6e248000a9c460f9aea8f1 100644
--- a/test/functional/p2p_ping.py
+++ b/test/functional/p2p_ping.py
@@ -25,12 +25,11 @@ class NodeNoPong(P2PInterface):
 
 
 class PingPongTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 1
-        # Set the peer connection timeout low. It does not matter for this
-        # test, as long as it is less than TIMEOUT_INTERVAL.
-        self.extra_args = [["-peertimeout=1"]]
+    setup_clean_chain = True
+    num_nodes = 1
+    # Set the peer connection timeout low. It does not matter for this
+    # test, as long as it is less than TIMEOUT_INTERVAL.
+    extra_args = [["-peertimeout=1"]]
 
     def check_peer_info(self, *, pingtime, minping, pingwait):
         stats = self.nodes[0].getpeerinfo()[0]
diff --git a/test/functional/p2p_sendheaders.py b/test/functional/p2p_sendheaders.py
index 6eb463ea9eb80e6dddad9f5178cef09ccec570d8..550f631dfb5bf67f8bb88375d35161c831247ec7 100644
--- a/test/functional/p2p_sendheaders.py
+++ b/test/functional/p2p_sendheaders.py
@@ -193,10 +193,9 @@ class BaseNode(P2PInterface):
 
 
 class SendHeadersTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 2
-        self.extra_args = [["-noparkdeepreorg"], ["-noparkdeepreorg"]]
+    setup_clean_chain = True
+    num_nodes = 2
+    extra_args = [["-noparkdeepreorg"], ["-noparkdeepreorg"]]
 
     def mine_blocks(self, count):
         """Mine count blocks and return the new tip."""
diff --git a/test/functional/p2p_timeouts.py b/test/functional/p2p_timeouts.py
index 5c02484b77e1b3530e9dc4e6bcff40d2e7d19906..93e3456a216f5a81802d47fb9323f14deaed84ea 100644
--- a/test/functional/p2p_timeouts.py
+++ b/test/functional/p2p_timeouts.py
@@ -30,11 +30,10 @@ class TestP2PConn(P2PInterface):
 
 
 class TimeoutsTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 1
-        # set timeout to receive version/verack to 3 seconds
-        self.extra_args = [["-peertimeout=3"]]
+    setup_clean_chain = True
+    num_nodes = 1
+    # set timeout to receive version/verack to 3 seconds
+    extra_args = [["-peertimeout=3"]]
 
     def mock_forward(self, delta):
         self.mock_time += delta
diff --git a/test/functional/p2p_tx_privacy.py b/test/functional/p2p_tx_privacy.py
index 5bf7310b0513ec2fb8b0b82bad33772a12a782e6..2699048690adfd7b3ddcf9add0fd75c4732616e2 100644
--- a/test/functional/p2p_tx_privacy.py
+++ b/test/functional/p2p_tx_privacy.py
@@ -44,8 +44,7 @@ class P2PTxSpy(P2PInterface):
 
 
 class TxPrivacyTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
+    num_nodes = 1
 
     def run_test(self):
         self.wallet = MiniWallet(self.nodes[0])
diff --git a/test/functional/p2p_unrequested_blocks.py b/test/functional/p2p_unrequested_blocks.py
index 07fa3c9ab4e8c910e619948af3128cbeb9475f06..e1b51a4cd04e57ef43f3c103ed9c1b93cb52d5e5 100644
--- a/test/functional/p2p_unrequested_blocks.py
+++ b/test/functional/p2p_unrequested_blocks.py
@@ -71,10 +71,9 @@ from test_framework.util import assert_equal, assert_raises_rpc_error
 
 
 class AcceptBlockTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 2
-        self.extra_args = [["-noparkdeepreorg"], ["-minimumchainwork=0x10"]]
+    setup_clean_chain = True
+    num_nodes = 2
+    extra_args = [["-noparkdeepreorg"], ["-minimumchainwork=0x10"]]
 
     def setup_network(self):
         self.setup_nodes()
-- 
2.50.1 (Apple Git-155)

